### PR TITLE
Add Custom Networks

### PR DESCRIPTION
**Pull request by FrederikBolding**
- Add possibility to add a custom network
- Fixed issues with node selection following Selector refactor

### CODE REVIEW

**Comment by FrederikBolding** [comment]
[ch6659]

File: src/features/Settings/components/AddOrEditNetworkNode.tsx [hunk 1]
```diff
@@ -254,19 +296,58 @@ export default function AddOrEditNetworkNode({
...
+                  baseUnit: baseUnit as TTicker,
+                  baseAsset: generateAssetUUID(chainId!),
+                  isCustom: true,
+                  assets: [],
+                  contracts: [],
+                  nodes: [node],
+                  dPaths: {
+                    [WalletId.TREZOR]: DPaths.ETH_TESTNET,
+                    [WalletId.LEDGER_NANO_S]: DPaths.ETH_TESTNET,
+                    [WalletId.MNEMONIC_PHRASE]: DPaths.ETH_TESTNET
```
Comment (GuillaumeRx) [inline src/features/Settings/components/AddOrEditNetworkNode.tsx:314]: Are you sure those DPaths are the one to use ?

File: src/features/Settings/components/AddOrEditNetworkNode.tsx [hunk 2]
```diff
@@ -254,19 +296,58 @@ export default function AddOrEditNetworkNode({
...
+              };
+              addAsset(baseAsset, network.baseAsset);
+              addNetwork(network);
             } else {
-              addNodeToNetwork(node, selectedNetworkId);
+              if (editNode) {
+                updateNode(node, selectedNetworkId, editNode.name);
+              } else {
+                addNodeToNetwork(node, selectedNetworkId);
+              }
```
Comment (GuillaumeRx) [inline src/features/Settings/components/AddOrEditNetworkNode.tsx:344]: Avoid nested if

File: src/features/Settings/components/AddOrEditNetworkNode.tsx [hunk 3]
```diff
@@ -6,13 +6,30 @@ import styled from 'styled-components';
...
+  GITHUB_RELEASE_NOTES_URL,
+  LETS_ENCRYPT_URL,
+  EXT_URLS,
+  DPathsList as DPaths
+} from '@config';
 import { NetworkUtils } from '@services/Store/Network';
 import { ProviderHandler } from '@services/EthService/network';
 
 import backArrowIcon from '@assets/images/icn-back-arrow.svg';
+import { generateAssetUUID } from '@utils';
```
Comment (GuillaumeRx) [inline src/features/Settings/components/AddOrEditNetworkNode.tsx:32]: Imports order

File: src/features/Settings/components/AddOrEditNetworkNode.tsx (hunk 1 above)
Comment (FrederikBolding) [reply src/features/Settings/components/AddOrEditNetworkNode.tsx:314]: Hmm, do we just want to use the default ones?

Comment (GuillaumeRx) [reply src/features/Settings/components/AddOrEditNetworkNode.tsx:314]: Let me check the one used with ganache

Comment (FrederikBolding) [reply src/features/Settings/components/AddOrEditNetworkNode.tsx:314]: Looks to be `m/44'/60'/0'/0/{account_index}` for Ganache. Which is `ETH_DEFAULT`

Comment (GuillaumeRx) [reply src/features/Settings/components/AddOrEditNetworkNode.tsx:314]: Yup ! You're right :) I've got the same

File: src/features/Settings/components/AddOrEditNetworkNode.tsx [hunk 4]
```diff
@@ -27,9 +27,9 @@ import {
 } from '@config';
 import { NetworkUtils } from '@services/Store/Network';
 import { ProviderHandler } from '@services/EthService/network';
+import { generateAssetUUID } from '@utils';
```
Comment (GuillaumeRx) [inline src/features/Settings/components/AddOrEditNetworkNode.tsx:30]: It should go higher, just after `@config` IMO

**Pull request merged**